### PR TITLE
Consolidate consecutive duplicate logger lines

### PR DESCRIPTION
**Pull request by StarbotArc**
Basically, no more big file sizes from repetitive logs. This pull request exists for issue #1291.

### CODE REVIEW

**Comment by nico-abram** [comment]
Why is last_msg a pointer? Can't we use a std::string directly and use an empty string instead of null?

**Comment by StarbotArc** [comment]
Sorry, I don't have crazy on-hand experience with C++, I actually write more C code, so I resorted more to pointers instead. If you think that's a better idea, I would say go for it.

**Comment by nico-abram** [comment]
LGTM

Only nits like I mentioned on discord are 

(1) maybe we can unlock the mutex before the call to plog by making a string copy
(2) I think we can remove the `if (this->last_msg.size() > 0)` condition, the initial empty string should work fine on the `if (message == this->last_msg)` check

File: src/Core/Misc/PlogLogger.cpp [hunk 1]
```diff
@@ -84,7 +86,19 @@ PlogLogger::PlogLogger() {
 }
 
 void PlogLogger::log(Core::ILogger::Severity logLevel, const std::string_view message) {
-    PLOG(PlogLogger::convertSeverity(logLevel)) << message;
+    const std::lock_guard<std::mutex> lock(this->mutex);
+    if (this->last_msg.empty()) {
```
Comment (poco0317) [inline src/Core/Misc/PlogLogger.cpp:90]: if last_msg is empty, then check to see if the new message is equal to that empty message? else if it isnt empty, set the last message to the new message? youve reversed the logic with the most recent commit

**Review by poco0317 — CHANGES_REQUESTED** [review]

File: src/Core/Misc/PlogLogger.cpp (hunk 1 above)
Comment (StarbotArc) [reply src/Core/Misc/PlogLogger.cpp:90]: Yeah, that's a major mistake, I'll get to fixing that.

**Review by poco0317 — APPROVED** [review]

**Comment by poco0317** [comment]
superceded by #1305

**Pull request closed without merging**